bullet/src/SDFFeatures.cc

```diff
@@ -0,0 +1,382 @@
+  baseTransform.setOrigin(convertVec(poseTranslation));
+  baseTransform.setBasis(convertMat(poseLinear));
+
+  // Create link
+  // (TO-DO: do we want to use MotionState?)
+  //  2nd part: Do motion state use the same transformation?
+  if (this->models.at(_modelID)->fixed)
+  {
+    mass = 0;
+    linkInertiaDiag = btVector3(0, 0, 0);
+  }
+
+  btDefaultMotionState* myMotionState = new btDefaultMotionState(baseTransform);
+  btCollisionShape* collisionShape = new btCompoundShape();
+  btRigidBody::btRigidBodyConstructionInfo
+    rbInfo(mass, myMotionState, collisionShape, linkInertiaDiag);
+  btRigidBody* body = new btRigidBody(rbInfo);
+  body->setActivationState(DISABLE_DEACTIVATION);
+
+  const auto &world = this->worlds.at(modelInfo->world)->world;
+
+  // Models collide with everything except themselves
+  const int modelCollisionGroup = 1 << this->collisionGroups.at(_modelID);
+  const int collisionMask = 0xFFFFFFFF & ~modelCollisionGroup;
+  world->addRigidBody(body, modelCollisionGroup, collisionMask);
+  // world->addRigidBody(body);
+
+  // Generate an identity for it
+  const auto linkIdentity = this->AddLink({name, body, _modelID, pose, mass,
+    linkInertiaDiag});
+  return linkIdentity;
+}
+
+Identity SDFFeatures::ConstructSdfCollision(
+    const Identity &_linkID,
+    const ::sdf::Collision &_collision)
+{
+  if (!_collision.Geom())
+  {
+    ignerr << "The geometry element of collision [" << _collision.Name() << "] "
+           << "was a nullptr\n";
+    return this->GenerateInvalidId();
+  }
+
+  const auto &geom = _collision.Geom();
+  btCollisionShape* shape = nullptr;
+
+  if (geom->BoxShape())
+  {
+    const auto box = geom->BoxShape();
+    const auto size = math::eigen3::convert(box->Size());
+    const auto halfExtents = convertVec(size)*0.5;
+    shape = new btBoxShape(halfExtents);
+  }
+  else if (geom->SphereShape())
+  {
+    const auto sphere = geom->SphereShape();
+    const auto radius = sphere->Radius();
+    shape = new btSphereShape(radius);
+  }
+  else if (geom->CylinderShape())
+  {
+    const auto cylinder = geom->CylinderShape();
+    const auto radius = cylinder->Radius();
+    const auto halfLength = cylinder->Length()*0.5;
+    shape = new btCylinderShapeZ(btVector3(radius, radius, halfLength));
+  }
+  else if (geom->PlaneShape())
+  {
+    const auto plane = geom->PlaneShape();
+    const auto normal = convertVec(math::eigen3::convert(plane->Normal()));
+    shape = new btStaticPlaneShape(normal, 0);
+  }
+
+  // TODO(lobotuerk) find if there's a  way to use friction and related
+  // info in bullet dynamics
+
+  const auto &surfaceElement = _collision.Element()->GetElement("surface");
+
+  // Get friction
+  const auto &odeFriction = surfaceElement->GetElement("friction")
+                              ->GetElement("ode");
+  const auto mu = odeFriction->Get<btScalar>("mu");
+  const auto mu2 = odeFriction->Get<btScalar>("mu2");
+  const auto mu3 = odeFriction->Get<btScalar>("mu3");
```

Comment:
SDF format does not have a default value for mu3, so you get 0 here if not defined on the .sdf. Having 0 on one of the forces axis does not seem good (and tests probe that the whole movement of the robot misbehaves with this)

Reply:
Is this still an issue? If it is, you could first check if `mu3` is valid.